src/core/server/http/http_config.ts

```diff
@@ -20,141 +21,143 @@ const hostURISchema = schema.uri({ scheme: ['http', 'https'] });
+          [
+            schema.arrayOf(hostURISchema, { minSize: 1 }),
+            schema.arrayOf(schema.literal('*'), { minSize: 1, maxSize: 1 }),
+          ],
+          {
+            defaultValue: ['*'],
+          }
+        ),
+      },
+      {
+        validate(value) {
+          if (value.allowCredentials === true && value.allowOrigin.includes('*')) {
+            return 'Cannot specify wildcard origin "*" with "credentials: true". Please provide a list of allowed origins.';
+          }
         },
-        {
-          validate(value) {
-            if (value.allowCredentials === true && value.allowOrigin.includes('*')) {
-              return 'Cannot specify wildcard origin "*" with "credentials: true". Please provide a list of allowed origins.';
-            }
-          },
+      }
+    ),
+    customResponseHeaders: schema.recordOf(schema.string(), schema.any(), {
+      defaultValue: {},
+    }),
+    host: schema.string({
+      defaultValue: 'localhost',
+      hostname: true,
+      validate(value) {
+        if (value === '0') {
+          return 'value 0 is not a valid hostname (use "0.0.0.0" to bind to all interfaces)';
         }
+      },
+    }),
+    maxPayload: schema.byteSize({
+      defaultValue: '1048576b',
+    }),
+    port: schema.number({
+      defaultValue: 5601,
+    }),
+    rewriteBasePath: schema.boolean({ defaultValue: false }),
+    ssl: sslSchema,
+    keepaliveTimeout: schema.number({
+      defaultValue: 120000,
+    }),
+    socketTimeout: schema.number({
+      defaultValue: 120000,
+    }),
+    compression: schema.object({
+      enabled: schema.boolean({ defaultValue: true }),
+      referrerWhitelist: schema.maybe(
+        schema.arrayOf(
+          schema.string({
+            hostname: true,
+          }),
+          { minSize: 1 }
+        )
+      ),
+    }),
+    uuid: schema.maybe(
+      schema.string({
+        validate: match(uuidRegexp, 'must be a valid uuid'),
+      })
+    ),
+    xsrf: schema.object({
+      disableProtection: schema.boolean({ defaultValue: false }),
+      allowlist: schema.arrayOf(
+        schema.string({ validate: match(/^\//, 'must start with a slash') }),
+        { defaultValue: [] }
       ),
-      customResponseHeaders: schema.recordOf(schema.string(), schema.any(), {
-        defaultValue: {},
-      }),
-      host: schema.string({
-        defaultValue: 'localhost',
-        hostname: true,
+    }),
+    requestId: schema.object(
+      {
+        allowFromAnyIp: schema.boolean({ defaultValue: false }),
+        ipAllowlist: schema.arrayOf(schema.ip(), { defaultValue: [] }),
+      },
+      {
         validate(value) {
-          if (value === '0') {
-            return 'value 0 is not a valid hostname (use "0.0.0.0" to bind to all interfaces)';
+          if (value.allowFromAnyIp === true && value.ipAllowlist?.length > 0) {
+            return `allowFromAnyIp must be set to 'false' if any values are specified in ipAllowlist`;
```

Comment:
The diff is misleading, the PR didn't change anything in the http config schema. Only effective diff in this file is the added `rename('maxPayloadBytes', 'maxPayload')` deprecation.